lib/TPP/MapToBatchReduceGEMM.cpp

```diff
@@ -80,11 +89,17 @@ static LogicalResult checkAccessPatterns(linalg::LinalgOp linalgOp) {
   SmallVector<AffineMap> compressedDimMaps = compressUnusedDims(maps);
   using MapList = ArrayRef<ArrayRef<AffineExpr>>;
   auto infer = [](MapList m) { return AffineMap::inferFromExprList(m); };
-  AffineExpr r1, p3, p4, r2;
-  bindDims(linalgOp.getContext(), r1, p3, p4, r2);
-  // Expected access patterns of BRGEMM
-  SmallVector<AffineMap> expectedMaps =
-      infer({{r1, p3, r2}, {r1, r2, p4}, {p3, p4}});
+  AffineExpr r1, p3, p4, r2, r3;
+  SmallVector<AffineMap> expectedMaps;
+  if (hasVNNIMap) {
+    bindDims(linalgOp.getContext(), r1, p3, p4, r2, r3);
+    // Expected access patterns of BRGEMM
+    expectedMaps =
+        infer({{r1, p3, r2}, {r1, r2.floorDiv(2), p4, r3}, {p3, p4}});
```

Comment:
We are going to always use 2?

Reply:
right now, we're only considering bf16 packing by a factor of 2. Therefore, I have a hardcoded 2 here. I can change this such that a variable is used, which is assigned 2 by default.

Reply:
Here there could be a `static always_inline getVNNIPackingRatio(Type)` or something that returns 2 for BF16 and 4 for BF8 (or something like that).

Reply:
Added getVNNIBlockingSize utility